x/upgrade/plan/info.go

```diff
@@ -15,24 +15,51 @@ import (
 
 // Info is the special structure that the Plan.Info string can be (as json).
 type Info struct {
+	parseConfig ParseConfig `json:"-"`
+
 	Binaries BinaryDownloadURLMap `json:"binaries"`
 }
 
 // BinaryDownloadURLMap is a map of os/architecture stings to a URL where the binary can be downloaded.
 type BinaryDownloadURLMap map[string]string
 
+// ParseConfig is used to configure the parsing of a Plan.Info string.
+type ParseConfig struct {
+	// EnforceChecksum, if true, will cause all downloaded files to be checked against their checksums.
+	// When false, checksums are not enforced to be present in the url.
+	EnforceChecksum bool
```

Comment:
if this is true and cosmovisor flag is false, which one takes priority?

Reply:
Basically, the software-upgrade CLI still requires a checksum for binary unless a flag is passed to disable it.
And cosmovisor by default do not require the software upgrade binary to have a checksum but an env variable can enforce it.